src/vivarium_gates_mncnh/components/children.py

```diff
@@ -14,127 +24,105 @@ class NewChildren(Component):
-            group_data = newborn_sex[newborn_sex == sex]
-            sex_exposure = self.exposure.loc[sex]
-            categorical_exposures.append(
-                self.randomness.choice(
-                    group_data.index,
-                    choices=sex_exposure.parameter.tolist(),
-                    p=sex_exposure.value.tolist(),
-                    additional_key="categorical_exposure",
-                )
-            )
-        categorical_exposures = pd.concat(categorical_exposures).sort_index()
-        return categorical_exposures
-
-    def _sample_continuous_exposure(self, categorical_exposure: pd.Series):
-        intervals = self.category_intervals.loc[categorical_exposure]
-        intervals.index = categorical_exposure.index
-        exposures = []
-        for axis in ["birth_weight", "gestational_age"]:
-            draw = self.randomness.get_draw(categorical_exposure.index, additional_key=axis)
-            lower, upper = intervals[f"{axis}_lower"], intervals[f"{axis}_upper"]
-            exposures.append((lower + (upper - lower) * draw).rename(axis))
-        return pd.concat(exposures, axis=1)
-
-    ################
-    # Data loading #
-    ################
-
-    def _get_category_intervals(self, builder: Builder):
-        categories = builder.data.load(data_keys.LBWSG.CATEGORIES)
-        category_intervals = pd.DataFrame(
-            data=[
-                (category, *self._parse_description(description))
-                for category, description in categories.items()
-            ],
-            columns=[
-                "category",
-                "birth_weight_lower",
-                "birth_weight_upper",
-                "gestational_age_lower",
-                "gestational_age_upper",
-            ],
-        ).set_index("category")
-        return category_intervals
-
-    @staticmethod
-    def _parse_description(description: str) -> Tuple:
-        birth_weight = [
-            float(val)
-            for val in description.split(", [")[1].split(")")[0].split("]")[0].split(", ")
-        ]
-        gestational_age = [
-            float(val)
-            for val in description.split("- [")[1].split(")")[0].split("+")[0].split(", ")
+        pop = self.population_view.get(event.index)
+        alive_children = pop.loc[pop[COLUMNS.CHILD_ALIVE] == "alive"]
+        # Update age of children to get correctlookup values - use midpoint of age groups
+        if self._sim_step_name() == SIMULATION_EVENT_NAMES.EARLY_NEONATAL_MORTALITY:
+            pop.loc[alive_children.index, COLUMNS.CHILD_AGE] = (7 / 2) / 365.0
+        else:
+            pop.loc[alive_children.index, COLUMNS.CHILD_AGE] = ((28 - 7) / 2) / 365.0
+
+        self.population_view.update(pop)
+
+
+class ChildrenBirthExposure(Component):
+    ##############
+    # Properties #
+    ##############
+
+    @property
+    def columns_created(self) -> list[str]:
+        return [
+            COLUMNS.GESTATIONAL_AGE,
```

Comment:
Do we still have multiple columns for gestational age and birth weight?

Reply:
Yes. One is the birth exposure columns that come from the pipeline and then these that are just gestational age and birth weight. I'm not sure why the NO project initialized these columns separately but I assume it is so we can put some effect on simulants and see if that changes these values from the pipelien.